kubernetes/structure_label_selector.go

```diff
@@ -50,6 +61,26 @@ func expandLabelSelector(l []interface{}) *metav1.LabelSelector {
 	return obj
 }
 
+func expandNamespaceSelector(n []interface{}) *metav1.LabelSelector {
```

Comment:
I don't think we want to return `*meta.LabelSelector` on  `expandNamespaceSelector`

Although they share the same characteristics we should be returning `NamespaceSelector`, i assume this can be found in the same `meta` object.

Reply:
leveraging metav1.LabelSelector as a temporary solution for handling namespace selection due to the absence of a dedicated NamespaceSelector type in the metav1 package.

Reply:
Ahhh i see that in the docs that it's marked as `LabelSelector` also:

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.31/#podaffinityterm-v1-core